src/common/transformations/src/transformations/common_optimizations/StateManagementPattern.cpp

```diff
@@ -0,0 +1,230 @@
+        auto k_reshape = std::make_shared<v1::Reshape>(k_transpose, v0::Constant::create(element::i32, Shape{1}, {0, 0, -1}), true);
+
+        std::shared_ptr<Node> v_transpose_order = kv_transpose_order; // eeeh, is it a right way to assign Constants? Maybe I should clone somehow?
+        if (pattern_map.find(v_order) != pattern_map.end()) { // reapply transpose found in the graph by manipulating of indices of our Transpose
+            v_transpose_order = std::make_shared<v1::Gather>(pattern_map.at(v_order), kv_transpose_order, v0::Constant::create(element::i32, Shape{}, {0}));
+        }
+        auto v_transpose = std::make_shared<v1::Transpose>(real_v, v_transpose_order);
+        auto v_reshape = std::make_shared<v1::Reshape>(v_transpose, v0::Constant::create(element::i32, Shape{1}, {0, 0, -1}), true);
+
+        // TODO: Detect whether SDPA in the model graph has `scale` argument set and use it instead of the computed scale below
+        // Most likely `scale` will always be a constant in real inference, but dynamic dimension propagation may not always derive it as a constant
+        // That's why a sub-graph computing `scale` is built instead of just a constant node.
+        auto hidden_shape = std::make_shared<v0::ShapeOf>(real_q);
+        auto hidden_dim = std::make_shared<v1::Gather>(hidden_shape, v0::Constant::create(element::i32, Shape{}, {-1}), v0::Constant::create(element::i32, Shape{}, {0}));
+        
+        auto scale = std::make_shared<v1::Divide>(v0::Constant::create(element::f32, Shape{}, {1}), std::make_shared<v0::Sqrt>(std::make_shared<v0::Convert>(hidden_dim, element::f32)));
+
+        std::shared_ptr<Node> alibi_slopes;
+        if (pattern_map.find(alibi) != pattern_map.end()) {
+            std::cout << "alibi slopes applied" << std::endl;
+            alibi_slopes = std::make_shared<v1::Reshape>(pattern_map.at(alibi), v0::Constant::create(element::i32, Shape{1}, {-1}), false); //here {-1} is interesting in Python TODO: discuss
+            if (alibi_slopes->get_element_type() == element::f32) {
+                alibi_slopes = std::make_shared<v0::Convert>(alibi_slopes, element::f32); //todo
+            }
+        } else {
+            alibi_slopes = v0::Constant::create(element::f32, Shape{1}, {}); // correctly created?
+        }
+
+        OutputVector params = {
+            q_reshape,
+            k_reshape,
+            v_reshape,
+            k_parameter,
+            v_parameter
+        };
+        params.insert(params.end(), model_remaining_params.begin(), model_remaining_params.end());
+        std::initializer_list<std::shared_ptr<Node>> additional_params = {scale, alibi_slopes, sliding_window};
+        params.insert(params.end(), additional_params.begin(), additional_params.end());
+
+        // Really not sure if I construct correctly because the Python code uses an additional function
+        auto paged_attention = std::make_shared<ov::op::PagedAttentionExtension>(params);
```

Comment:
TODO: The python code uses an additional arguments_as_outputs() function, but I don't know if we need it here.

Reply:
removed.